### PR TITLE
Set the bazel output base directory to a well known location

### PR DESCRIPTION
**Pull request by rmohr**
Ensure that we know which directory bazel will use as a cache
directories, independent of the user used on the prow job.

Further mounting an emptyDir at this well known location if jobs decide
to run bazel unnested.

This way bazel will not write massively into the container overlay,
which can fill up the container space pretty fast and which can also
perform bad.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

**Comment by rmohr** [comment]
bazelrc file when the cache is enabled:

```
# cat /etc/bazel.bazelrc 
startup --output_user_root=/tmp/cache/bazel
startup --host_jvm_args=-Dbazel.DigestFunction=sha256
build --remote_local_fallback
build --remote_http_cache=http://bazel-cache.kubevirt-prow.svc.cluster.local:8080///workspace,f107c644ec3d43330ad677d00bcd326f
```

bazelrc when the cache is not requested:

```
# cat /etc/bazel.bazelrc 
startup --output_user_root=/tmp/cache/bazel
```

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve
/hold

Great thx, feel free to unhold when ready.

**Comment by rmohr** [comment]
@fgimenez improved the weird wording on one of the comments, should be good otherwise. Would need another lgtm.

**Pull request merged**
